### PR TITLE
AI Assistant: Add and use ImproveToolbarDropdownMenu in block toolbar

### PR DESCRIPTION
**Pull request by retrofox**
This PR adds and uses the `<ImproveToolbarDropdownMenu />` component.

Fixes #

## Proposed changes:

* AI Assistant: Add and use ImproveDropdown in block toolbar

### Other information:

- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Go to the block editor
* Create and use the AI Assistant block
* Confirm the Improve dropdown works as expected
  * `Make longer`, `Make shorter`,  `Summarize`
  * Confirm when it's editing the post title, the `Summarize` option is not listed

image

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/ai-assistant/components/improve-dropdown/index.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+			} }
+		>
+			{ () => {
+				// Exclude quick edits from the list.
+				const quickActionsListFiltered = quickActionsList.filter(
+					quickAction => ! exclude.includes( quickAction.key )
+				);
+
+				return (
+					<MenuGroup label={ __( 'Improve', 'jetpack' ) }>
```
Comment (renatoagds) [inline projects/plugins/jetpack/extensions/blocks/ai-assistant/components/improve-dropdown/index.tsx]: IMO we should give more context like we do in `Write with AI` ... Something like `Improve based on previous context`.

Comment (retrofox) [reply projects/plugins/jetpack/extensions/blocks/ai-assistant/components/improve-dropdown/index.tsx]: Let's don't add a menu group label for now? Same as now.

Comment (retrofox) [reply projects/plugins/jetpack/extensions/blocks/ai-assistant/components/improve-dropdown/index.tsx]: image

**Review by dhasilva — APPROVED** [review]
LGTM

**Pull request merged**